RichPushNotifications/NotificationService.swift

```diff
@@ -0,0 +1,102 @@
+import UIKit
```

Comment:
This is almost 100% copy-pasted from Braze's example code. The only difference is that I changed `print` to my own `printDebug` method to keep us from spamming the console in production. I gave this a read-through and I think this implementation is fine, if a bit verbose and cautious.